### PR TITLE
Remove identity core classes from public api

### PR DESCRIPTION
**Pull request by antrix1989**
## Proposed changes

We shouldn't have IdentityCore classes in the public API of MSAL. 

## Type of change

- [ ] Feature work
- [x] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: MSAL/MSAL.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -147,6 +147,12 @@
 		1EF395FF246DFAD200647FDB /* MSALAuthScheme.m in Sources */ = {isa = PBXBuildFile; fileRef = 1EF395FC246DFAD200647FDB /* MSALAuthScheme.m */; };
```
Comment (kaisong1990) [inline MSAL/MSAL.xcodeproj/project.pbxproj:147]: This pull request does not update CHANGELOG.md.

Please consider if this change would be noticeable to a partner or user and either update CHANGELOG.md or resolve this conversation.

**Review by ameyapat — APPROVED** [review]
@antrix1989 Does the authScheme parameter in below method in MSALErrorConvertor.m need to also conform to MSALAuthenticationSchemeProtocolInternal? 
``` 
+ (NSError *)msalErrorFromMsidError:(NSError *)msidError
                     classifyErrors:(BOOL)shouldClassifyErrors
                 msalOauth2Provider:(MSALOauth2Provider *)oauth2Provider
                            correlationId:(NSUUID *)correlationId
                         authScheme:(id<MSALAuthenticationSchemeProtocol>)authScheme
                         popManager:(MSIDDevicePopManager *)popManager
```

File: MSAL/src/MSALAuthenticationSchemePop+Internal.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// THE SOFTWARE.  
+
+
+#import <MSAL/MSAL.h>
+#import "MSALAuthenticationSchemePop.h"
+#import "MSALAuthenticationSchemeProtocolInternal.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSALAuthenticationSchemePop () <MSALAuthenticationSchemeProtocolInternal>
```
Comment (hieunguyenmsft) [inline MSAL/src/MSALAuthenticationSchemePop+Internal.h:32]: Just curious do you think we need `MSALAuthenticationSchemePop` available to public API? Or we can make it internal only and move back your changes from '`MSALAuthenticationSchemeProtocolInternal`' to '`MSALAuthenticationSchemeProtocol`'
I think client doesn't need to know about `MSALAuthenticationSchemePop`, they only care about the `MSALResult` object we return them. `authenticationScheme` in that object is NSString, so we don't really need '`MSALAuthenticationSchemePop`' available to public API. What do you think?

**Review by oldalton — APPROVED** [review]

File: MSAL/src/MSALAuthenticationSchemePop+Internal.h (hunk 2 above)
Comment (antrix1989) [reply MSAL/src/MSALAuthenticationSchemePop+Internal.h:32]: I think we need `initWithHttpMethod:requestUrl:nonce:additionalParameters` in public API to support POP, don't we? If you think it is not needed, let's discuss it.

**Comment by antrix1989** [comment]
> @antrix1989 Does the authScheme parameter in below method in MSALErrorConvertor.m need to also conform to MSALAuthenticationSchemeProtocolInternal?
> 
> ```
> + (NSError *)msalErrorFromMsidError:(NSError *)msidError
>                      classifyErrors:(BOOL)shouldClassifyErrors
>                  msalOauth2Provider:(MSALOauth2Provider *)oauth2Provider
>                             correlationId:(NSUUID *)correlationId
>                          authScheme:(id<MSALAuthenticationSchemeProtocol>)authScheme
>                          popManager:(MSIDDevicePopManager *)popManager
> ```

We don't use methods of MSALAuthenticationSchemeProtocolInternal there, so it is not needed.

File: MSAL/src/MSALAuthenticationSchemePop+Internal.h (hunk 2 above)
Comment (hieunguyenmsft) [reply MSAL/src/MSALAuthenticationSchemePop+Internal.h:32]: Ah you're right, we need it to create the signed header. So if the client knows they want AT Pop, they will use that class to get back AT Pop. Never mind my question earlier.

**Pull request merged**